src/connectors/autocomplete/connectAutocomplete.ts

```diff
@@ -0,0 +1,171 @@
+  currentRefinement: string;
+  indices: AutocompleteIndex[];
+  instantSearchInstance: InstantSearch;
+  refine: RefineFunction;
+}
+
+export type AutocompleteRenderer<TAutocompleteWidgetParams> = Renderer<
+  AutocompleteRendererOptions<
+    AutocompleteConnectorParams & TAutocompleteWidgetParams
+  >
+>;
+
+export type AutocompleteWidgetFactory<
+  TAutocompleteWidgetParams
+> = WidgetFactory<AutocompleteConnectorParams & TAutocompleteWidgetParams>;
+
+export type AutocompleteConnector = <TAutocompleteWidgetParams>(
+  render: AutocompleteRenderer<TAutocompleteWidgetParams>,
+  unmount?: Unmounter
+) => AutocompleteWidgetFactory<TAutocompleteWidgetParams>;
+
+const connectAutocomplete: AutocompleteConnector = (
+  renderFn,
+  unmountFn = noop
+) => {
+  checkRendering(renderFn, withUsage());
+
+  return widgetParams => {
+    const { escapeHTML = true } = widgetParams || {};
+
+    warning(
+      !(widgetParams as any).indices,
+      `Since InstantSearch.js 4, \`connectAutocomplete\` infers the indices from the tree of widgets.
+The \`indices\` option is ignored.`
+    );
+
+    type ConnectorState = {
+      instantSearchInstance?: InstantSearch;
+      refine?: RefineFunction;
+    };
+
+    const connectorState: ConnectorState = {};
+
+    return {
+      $$type: 'ais.autocomplete',
+
+      getConfiguration(previousParameters) {
+        const parameters = {
+          query: '',
+        };
+
+        if (!escapeHTML) {
+          return previousParameters.setQueryParameters(parameters);
+        }
+
+        return previousParameters.setQueryParameters({
+          ...parameters,
+          ...TAG_PLACEHOLDER,
+        });
+      },
+
+      init({ instantSearchInstance, helper }) {
+        connectorState.instantSearchInstance = instantSearchInstance;
+        connectorState.refine = (query: string) =>
+          helper.setQuery(query).search();
+
+        renderFn(
+          {
+            widgetParams,
+            currentRefinement: helper.state.query || '',
+            indices: [],
+            refine: connectorState.refine,
+            instantSearchInstance: connectorState.instantSearchInstance,
+          },
+          true
+        );
+      },
+
+      render({ helper, scopedResults }) {
+        const indices = scopedResults.map(scopedResult => {
+          scopedResult.results.hits = escapeHTML
+            ? escapeHits(scopedResult.results.hits)
+            : scopedResult.results.hits;
+
+          return {
+            index: scopedResult.results.index,
+            hits: scopedResult.results.hits,
+            results: scopedResult.results,
+          };
+        });
+
+        renderFn(
+          {
+            widgetParams,
+            currentRefinement: helper.state.query || '',
+            indices,
+            refine: connectorState.refine!,
+            instantSearchInstance: connectorState.instantSearchInstance!,
+          },
+          false
+        );
+      },
+
```

Comment:
The widget does not implement the routing lifecycle, we can add them in a separate PR though.